toolchain/check/sem_ir_diagnostic_converter.cpp

```diff
@@ -23,7 +37,7 @@ auto SemIRDiagnosticConverter::ConvertLoc(SemIRLoc loc,
                  "If we get invalid locations here, we may need to more "
                  "thoroughly track ImportDecls.");
 
-    DiagnosticLoc in_import_loc;
+    std::pair<DiagnosticLoc, int32_t> in_import_loc;
     auto import_loc_id = cursor_ir->insts().GetLocId(import_ir.decl_id);
     if (import_loc_id.is_node_id()) {
       // For imports in the current file, the location is simple.
```

Comment:
It looks like the calls to `ConvertLocInFile` here can produce byte offsets into files other than the current source file. In that case maybe we should set the byte offset to -1 and allow `ConvertLoc` to pick a better one?

Reply:
Done in ConvertLoc